src/rust/engine/options/src/parse.rs

```diff
@@ -274,49 +278,188 @@ fn format_parse_error(
     ))
 }
 
-#[allow(dead_code)]
-pub(crate) fn parse_bool(value: &str) -> Result<bool, ParseError> {
-    option_value_parser::bool(value).map_err(|e| format_parse_error("bool", value, e))
+pub(crate) fn parse_dict(value: &str) -> Result<DictEdit, ParseError> {
+    option_value_parser::dict_edit(value).map_err(|e| format_parse_error("dict", value, e))
+}
+
+pub(crate) trait Parseable: Sized {
+    fn parse(value: &str) -> Result<Self, ParseError>;
+    fn parse_list(value: &str) -> Result<Vec<ListEdit<Self>>, ParseError>;
 }
 
-#[allow(dead_code)]
-pub(crate) fn parse_int(value: &str) -> Result<i64, ParseError> {
-    option_value_parser::int(value).map_err(|e| format_parse_error("int", value, e))
+impl Parseable for bool {
+    fn parse(value: &str) -> Result<bool, ParseError> {
+        option_value_parser::bool(value).map_err(|e| format_parse_error("bool", value, e))
+    }
+
+    fn parse_list(value: &str) -> Result<Vec<ListEdit<bool>>, ParseError> {
+        option_value_parser::bool_list_edits(value)
+            .map_err(|e| format_parse_error("bool list", value, e))
+    }
 }
 
-#[allow(dead_code)]
-pub(crate) fn parse_float(value: &str) -> Result<f64, ParseError> {
-    option_value_parser::float(value).map_err(|e| format_parse_error("float", value, e))
+impl Parseable for i64 {
+    fn parse(value: &str) -> Result<i64, ParseError> {
+        option_value_parser::int(value).map_err(|e| format_parse_error("int", value, e))
+    }
+
+    fn parse_list(value: &str) -> Result<Vec<ListEdit<i64>>, ParseError> {
+        option_value_parser::int_list_edits(value)
+            .map_err(|e| format_parse_error("int list", value, e))
+    }
 }
 
-#[allow(dead_code)]
-pub(crate) fn parse_quoted_string(value: &str) -> Result<String, ParseError> {
-    option_value_parser::quoted_string(value).map_err(|e| format_parse_error("string", value, e))
+impl Parseable for f64 {
+    fn parse(value: &str) -> Result<f64, ParseError> {
+        option_value_parser::float(value).map_err(|e| format_parse_error("float", value, e))
+    }
+
+    fn parse_list(value: &str) -> Result<Vec<ListEdit<f64>>, ParseError> {
+        option_value_parser::float_list_edits(value)
+            .map_err(|e| format_parse_error("float list", value, e))
+    }
+}
+
+impl Parseable for String {
+    fn parse(value: &str) -> Result<String, ParseError> {
+        Ok(value.to_owned())
+    }
+
+    fn parse_list(value: &str) -> Result<Vec<ListEdit<String>>, ParseError> {
+        option_value_parser::string_list_edits(value)
+            .map_err(|e| format_parse_error("string list", value, e))
+    }
+}
+
+pub(crate) trait ListMember: Parseable + DeserializeOwned {}
```

Comment:
This is just a tagging trait for list members, to ensure they are also deserializable into an owned instance by serde, when we read them from json/yaml fromfiles.

Reply:
It was #1. 

But this code evolved while I was working on it, and now that I look at where it ended up, ListMember isn't actually needed anyway. Every Parseable can be a list member (because of its parse_list()), so it's easier to make Parseable a subtrait of DeserializeOwned and be done with it.

Good call!